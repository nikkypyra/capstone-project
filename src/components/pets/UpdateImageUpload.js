import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import AddButton from '../buttons/AddButton'

UpdateImageUpload.propTypes = {
  petImage: PropTypes.object.isRequired,
  handleUpload: PropTypes.func.isRequired,
  onClick: PropTypes.func,
  loading: PropTypes.number,
}

export default function UpdateImageUpload({
  petImage,
  handleUpload,
  onClick,
  loading,
}) {
  return (
    <>
      <ImageWrapper>
        <img src={petImage.imageUrl} alt="" />
      </ImageWrapper>
      <UploadWrapper>
        {loading > 0 && <span>Upload is {Math.trunc(loading)}% done</span>}
        <AddButton
          className="upload-button"
          text="Upload photo"
          onClick={onClick}
        />
        <input
          type="file"
          accept="image/*"
          className="file-input"
          onChange={handleUpload}
        />
      </UploadWrapper>
    </>
  )
}

const UploadWrapper = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  .file-input {
    opacity: 0;
    position: absolute;
  }
  button {
    margin-top: 8px;
  }
`

const ImageWrapper = styled.div`
  text-align: center;
  img {
    height: 200px;
    width: 200px;
    border: 8px solid var(--tertiary);
    border-radius: 50%;
  }
`
