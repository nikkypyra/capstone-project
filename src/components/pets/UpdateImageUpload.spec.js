import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import UpdateImageUpload from './UpdateImageUpload'

afterEach(cleanup)

const petImageMock = {}
const handleUpload = jest.fn()
const onClick = jest.fn()

describe('<UpdateImageUpload />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <UpdateImageUpload
          petImage={petImageMock}
          handleUpload={handleUpload}
          onClick={onClick}
        />
      </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
