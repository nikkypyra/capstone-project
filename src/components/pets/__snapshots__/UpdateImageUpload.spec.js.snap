// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateImageUpload /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="UpdateImageUpload__ImageWrapper-sc-1cin8yf-1 kOPpT"
  >
    <img
      alt=""
    />
  </div>
  <section
    class="UpdateImageUpload__UploadWrapper-sc-1cin8yf-0 gdHUqm"
  >
    <button
      class="sc-AxjAm eirFEf"
    >
      Upload photo
    </button>
    <input
      accept="image/*"
      class="file-input"
      type="file"
    />
  </section>
</DocumentFragment>
`;
